### ISSUE TITLE
Spfx 1.7 Linting

### ISSUE THREAD

**Issue by dkorolev1** [SharePoint/sp-dev-docs#3542]
#### Category
- [x ] Question
- [ ] Typo
- [ ] Bug
- [ ] Additional article idea

Hi,

I have a simple SPFx project that shows some tslint warnings when building. I would like to deal with these warnings, but I don’t understand some things:

1. The tslint.json file located in the root of my project. But, when I make any changes in this file (for example, disable some rules that cause warnings) it doesn’t affect anything. I see these warnings as before. As if another tslint.json file is used. For example, I added the following rule: `"no-any": false`, but I still have the warnings like 
`error no-any: Type declaration of 'any' loses type-safety. Consider replacing it with a more precise type,`

2. I've installed the TSLint extension to my VS Code, but it doesn't highlight any warnings on my code. I can see these warnings only after building my solution in the Output window.

Thanks.

**Comment by andrewconnell** [comment]
Please provide additional context, specifically SPFx version & Node.js version.

Also would help if you provided more detail on your warnings. Creating a default project with SPFx v1.7.1, adding a single `private something: any;` member to the default web part project and building yields no warnings from tslint... thus I can't repro...

**Comment by dkorolev1** [comment]
Hi @andrewconnell,

Sorry for the delay. So, here are the main details:

SPFx version: 1.7
Node.js version: 8.14.0

? What is your solution name? **linting-test**
? Which baseline packages do you want to target for your component(s)? **SharePoint 2019 onwards, including SharePoint Online**
? Where do you want to place the files? **Use the current folder**
Found npm version 6.4.1
? Do you want to allow the tenant admin the choice of being able to deploy the solution to all sites immediately without
 running any feature deployment or adding apps in sites? **Yes**
? Which type of client-side component to create? **WebPart**
Add new Web part to solution linting-test.
? What is your Web part name? **HelloWorld**
? What is your Web part description? **HelloWorld description**
? Which framework would you like to use? **React**

So, I created a new project in accordance with the configuration below, added a new property to the web part class (`private something: any;`) and a rule (`"no-any": false`) to the tslint.json. And that's what I get when I run  `gulp build`:

_Warning - tslint - src/webparts/helloWorld/HelloWorldWebPart.ts(19,22): error no-any: Type declaration of 'any' loses type-safety. Consider replacing it with a more precise type, the empty type ('{}'), or suppress this occurrence._

Here is a [link](https://github.com/dkorolev1/linting-test)  to the test repo. 
Thanks!

**Comment by mcmynn83** [comment]
If you want to just disable the no any check on a single line you can add the follow tslint comment:
// tslint:disable-next-line:no-any

**Comment by VesaJuvonen** [comment]
Closing as answered by engineering for now, unless we need to further discussion of this.

**Comment by andrewconnell** [comment]
Quick follow up... @dkorolev1 try using the latest version of SPFx generator. I see you're using 1.7, but the later versions such as 1.8 are not showing the same results.

**Comment by dkorolev1** [comment]
@mcmynn83 , Yes, I know, but it is not an option to add these comments everywhere. @andrewconnell, ok, I will try. Thanks to all!

**Comment by dkorolev1** [comment]
@andrewconnell , unfortunately, there are other troubles in SPFx 1.8 generator. Details: [3730](https://github.com/SharePoint/sp-dev-docs/issues/3730)

**Comment by andrewconnell** [comment]
@dkorolev1 Correct, but if you look in that issue, you'll see it was closed as a dupe to another issue. The work-around is documented until 1.8.1 is released. The fix is quite easy until 1.8.1 is released (I'd expect it in the next few weeks).

**Comment by dkorolev1** [comment]
@andrewconnell, yeah, I saw, thanks.

**Comment by dharnen** [comment]
@andrewconnell  

Hello, I'm experiencing the same issue as @dkorolev1 where any changes tot the rules listed in the tslint.json file don't seem to have any affect, except I don't have the option of upgrading to a newer spfx version. My client is running sp2016 on prem which locks you into 1.1.0. Is there a known work around/fix/explanation for why changes to the tslint.json file in the root are not reflected when serving up the solution?

Thanks!
